src/nvim/eval.c

```diff
@@ -5138,8 +5112,8 @@ bool garbage_collect(bool testing)
 
   // Jobs
   {
-    TerminalJobData *data;
-    map_foreach_value(jobs, data, {
+    Channel *data;
+    map_foreach_value(channels, data, {
       set_ref_in_callback(&data->on_stdout, copyID, NULL, NULL);
```

Comment:
Missing cast `(Callback *)` to hide `-Wincompatible_pointer_types` warning for first argument. 

Perhaps also add somewhere a `static_assert` to make sure that the `Callback` object is the first member in `CallbackReader`:
```c 
#include "nvim/assert.h"
#include <stddef.h>
STATIC_ASSERT(offsetof(CallbackReader,cb)== 0,"...");
```